### PR TITLE
Comment action finished

### PR DESCRIPTION
**Pull request by Onebu**
**POST api/posts/{postId}/reply + postBody** to reply

**GET api/posts/detail/{postId}** to get the post and its parent and childs

### CODE REVIEW

**Comment by Onebu** [comment]
I-23 finished
Added report function : 
http://localhost:8000/api/users/report/{post_id}/ to report a post,

### BTW I only added the filter of reported post in Search post and user's related post . 

        user = self.queryset.get(pk=request.user.pk)
        reported = user.reported.all().values_list('id')
        filtered = posts.all().exclude(id__in=reported)
Here is an example about how to filter post list to return in Views.py.

File: piopio/piopio_be/views.py [hunk 1]
```diff
@@ -299,21 +324,55 @@ def search(self, request):
...
+            headers = self.get_success_headers(serializer.data)
+            child_id = serializer.data['id']
+            tmp = self.get_queryset().filter(id=child_id)
+            tmp.get().parent.add(parent.get().id)
+            tmp.get().save()
+            return Response(serializer.data, status=status.HTTP_201_CREATED, headers=headers)
+
+        except ValueError:
+            return Response({'message': 'Something went wrong!'}, status.HTTP_404_NOT_FOUND)
+        except models.User.DoesNotExist:
```
Comment (smontode24) [inline piopio/piopio_be/views.py]: Change to models.Post.DoesNotExist

File: piopio/piopio_be/views.py [hunk 2]
```diff
@@ -178,13 +178,19 @@ def related(self, request, userpk):
...
             user_id__in=related).order_by('-created_at')
         posts = add_likes_and_retweets(posts, userpk)
 
-        page = self.paginate_queryset(posts)
+        """
+        Filter the results with user's report list
+        """
+        user = self.queryset.get(pk=request.user.pk)
+        reported = user.reported.all().values_list('id')
+        filtered = posts.all().exclude(id__in=reported)
```
Comment (smontode24) [inline piopio/piopio_be/views.py:190]: The posts should also be filtered out if the post has more than 10 reports in total.

File: piopio/piopio_be/views.py [hunk 3]
```diff
@@ -299,21 +324,55 @@ def search(self, request):
...
+            page, many=True)
         return self.get_paginated_response(serialized_posts.data)
 
+    @action(methods=['GET'], detail=False, url_path="details/(?P<postpk>[^/.]+)", url_name="detail_post")
+    def details(self, request, postpk):
+        try:
+            post = self.get_queryset().filter(id=postpk)
+            parent = post.get().parent.all()
+            child = self.get_queryset().filter(parent__id__exact=postpk)
+            detail = list(chain(parent, post, child))
```
Comment (smontode24) [inline piopio/piopio_be/views.py]: Union of querysets without casting to list

**Review by smontode24 — CHANGES_REQUESTED** [review]
There are things missing (+ things marked in the code):

From the comment US:
- Modify api/posts/<id> endpoint to return the post with that id, its parent and its children (its comments). For now we will only return the children of the post (only the level behind that node in the tree).

From the report US:
- If one post has more than 10 reports should be filtered out in every view available, for that filter with custom manager overriding the all() method to not retrieve the ones reported.

+ Add README for every new/updated endpoint.

File: piopio/piopio_be/views.py (hunk 1 above)
Comment (Onebu) [reply piopio/piopio_be/views.py]: Fixed

File: piopio/piopio_be/views.py (hunk 3 above)
Comment (Onebu) [reply piopio/piopio_be/views.py]: Changed it to union.

File: piopio/piopio_be/views.py (hunk 2 above)
Comment (Onebu) [reply piopio/piopio_be/views.py:190]: I'll keep this part, the simple reason is that by now we only have to implement a "fake" report function (Also I'd like to point out that adding a counter to the report function will make the demo way more difficult cuz we need to log in 10 different account to report the same post).

**Comment by Onebu** [comment]
About the comment "tree" , I think we also need the parent comment to just make the ``back`` button to go to the parent level.

**Pull request merged**